### PR TITLE
Caching mne.get_datatypes to speed up _get_ignore_datatypes

### PR DESCRIPTION
**Pull request by apmellot**
### Before merging …

- [ ] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

**Review by hoechenberger — COMMENTED** [review]
Looks good! This performance improvement deserves to be credited in our changelog, don't you think? :)

**Comment by apmellot** [comment]
Like that? It's the first time I update a changelog haha

File: docs/source/v1.2.md.inc [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ### :new: New features & enhancements
 
+- mne.get_datatype is now cached to reduce computation time when large datasets are being processed. (#720 by @apmellot)
```
Comment (hoechenberger) [inline docs/source/v1.2.md.inc:5]: Very nice! I always like to have things a little bit less technical so users will get a feel for whether a change is going to help them or not. How about this:

```suggestion
- We improved caching to reduce initialization time when large datasets are being processed. (#720 by @apmellot)
```

**Review by hoechenberger — APPROVED** [review]
Thanks @apmellot 

I'll let you merge @agramfort

**Pull request merged**